### PR TITLE
 Make validation more robust for reindex TINC tests

### PR DESCRIPTION
**Pull request by asimrp**
To validate that a reindex operation indeed created a new index, the tests need
to inspect relfilenode values of the index's entry in pg_class.  Previously,
the GUC gp_select_invisible was turned on to get relfilenode prior to reindex.
Use of the GUC made the tests fickle.  E.g. running an unrelated test that
creates and drops a table with identical name as one of the index names in
reindex test is suffice to make the reindex tests fail.

Found several redundant tests, removed them on the way.

The reindex suite has many redundant tests.  Only valid scenarios need to be moved to isolation2 and the suite deleted from TINC.  For now, this PR addresses the cs-storage job failure in `gpdb_master` pipeline.  The patch will be backported to 5X_STABLE.

### CODE REVIEW

File: src/test/tinc/tincrepo/mpp/gpdb/tests/storage/lib/sql_isolation_testcase.py [hunk 1]
```diff
@@ -31,7 +31,10 @@
 class SQLIsolationExecutor(object):
     def __init__(self, dbname=''):
         self.processes = {}
-        self.command_pattern = re.compile(r"^(\d+)([&\\<\\>Uq]?)\:(.*)")
+        # The re.S flag makes the "." in the regex match newlines.
+        # When matched against a command in process_command(), all
+        # lines in the command are matched and sent as SQL query.
+        self.command_pattern = re.compile(r"^(\d+)([&\\<\\>Uq]?)\:(.*)", re.S)
```
Comment (ashwinstar) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/storage/lib/sql_isolation_testcase.py]: Nice. While moving these tests to regress isolation2 we should get this as well in framework there.

File: src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/reindex/concurrency/expected/reindextable_while_reindex_idx_ao_bitmap.ans [hunk 2]
```diff
@@ -1,10 +1,12 @@
 -- @Description Ensures that a reindex table during reindex index operations is ok
--- 
+--
 
 DELETE FROM reindex_crtab_ao_bitmap WHERE a < 128;
 DELETE 254
 1: BEGIN;
 BEGIN
+1: create temp table old_relfilenodes as (select gp_segment_id as dbid, relfilenode, oid, relname from gp_dist_random('pg_class') where relname = 'idx_reindex_crtab_ao_bitmap' union all select gp_segment_id as dbid, relfilenode, oid, relname from pg_class where relname = 'idx_reindex_crtab_ao_bitmap');
+CREATE 3
```
[the remark this answers is not in the document]
Comment (asimrp) [reply src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/reindex/concurrency/expected/reindextable_while_reindex_idx_ao_bitmap.ans]: You mean if the test is run against a GPDB with more than two segments?  Specific cluster config of 2 segments is already assumed elsewhere in the tests.

File: src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/reindex/concurrency/expected/reindextable_while_reindex_idx_ao_part_btree.ans [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 -- @Description Ensures that a reindex table during reindex index operations is ok
--- 
+--
 
 DELETE FROM reindex_crtab_part_ao_btree  WHERE id < 128;
 DELETE 254
 1: BEGIN;
 BEGIN
+1: create temp table old_relfilenodes as (select gp_segment_id as dbid, relfilenode, oid, relname from gp_dist_random('pg_class') where relname like 'idx_reindex_crtab_part_ao_btree%' union all select gp_segment_id as dbid, relfilenode, oid, relname from pg_class where relname like 'idx_reindex_crtab_part_ao_btree%');
+CREATE 9
```
[the remark this answers is not in the document]
Comment (asimrp) [reply src/test/tinc/tincrepo/mpp/gpdb/tests/storage/vacuum/reindex/concurrency/expected/reindextable_while_reindex_idx_ao_part_btree.ans]: That's right, three segments (2 primary and 1 master), each having three indexes, that comes out to 9.

**Comment by asimrp** [comment]
Thank you for the reviews.  @Chibin @cjcjameson the documentation is admittedly not precise.  When migrating to isolation2, we will describe the scenarios clearly.

**Pull request merged**